File: code/ch14/14.3.1_240.search-a-2d-matrix-ii.java
class Solution {
  public boolean searchMatrix(int[][] matrix, int target) {
      for (int i = 0; i < matrix.length; i++) {
          for (int j = 0; j < matrix[0].length; j++) {
              if (matrix[i][j] == target) {
                  return true;
              }
          }
      }

      return false;
  }
}

File: code/ch14/14.3.1_240.search-a-2d-matrix-ii.py
from typing import List
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix or not matrix[0]:
            return False

        m = len(matrix)
        n = len(matrix[0])

        for i in range(m):
            for j in range(n):
                if matrix[i][j] == target:
                    return True

        return False
